aptos-move/e2e-move-tests/src/tests/transaction_context.rs

```diff
@@ -0,0 +1,263 @@
+        .unwrap();
+
+    txn_ctx_store.max_gas_amount
+}
+
+#[test]
+fn test_transaction_context_sender() {
+    let mut harness = MoveHarness::new();
+    let account = setup(&mut harness);
+
+    let addr = call_get_sender_from_native_txn_context(&mut harness, &account);
+    println!("addr: {:?}", addr);
+    assert_eq!(addr, AccountAddress::ONE);
+}
+
+#[test]
+fn test_transaction_context_max_gas_amount() {
+    let mut harness = MoveHarness::new();
+    let account = setup(&mut harness);
+
+    let max_gas_amount = call_get_max_gas_amount_from_native_txn_context(&mut harness, &account);
+    println!("max_gas_amount: {:?}", max_gas_amount);
+    assert_eq!(max_gas_amount, 2000000);
+}
+
+#[test]
+fn test_transaction_context_fee_payer_none() {
+    let mut harness = MoveHarness::new();
+    let account = setup(&mut harness);
+
+    let fee_payer_opt = call_get_fee_payer_opt_from_native_txn_context(&mut harness, &account);
+    println!("fee_payer_opt: {:?}", fee_payer_opt);
+    assert_eq!(fee_payer_opt, None);
+}
+
+#[test]
+fn test_transaction_context_secondary_signers_empty() {
+    let mut harness = MoveHarness::new();
+    let account = setup(&mut harness);
+
+    let secondary_signers = call_get_secondary_signers_from_native_txn_context(&mut harness, &account);
+    println!("secondary_signers: {:?}", secondary_signers);
+    assert_eq!(secondary_signers, vec![]);
+}
+
+#[test]
+fn test_transaction_context_fee_payer_some() {
+    let mut harness = MoveHarness::new_with_features(
+        vec![
+            FeatureFlag::GAS_PAYER_ENABLED,
+            FeatureFlag::SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION,
+        ],
+        vec![],
+    );
+
+    let alice = setup(&mut harness);
+    let bob = harness.new_account_with_balance_and_sequence_number(1000000, 0);
+
+    let fun: MemberId = str::parse("0x1::transaction_context_test::store_fee_payer_opt_from_native_txn_context")
+        .unwrap();
+    let MemberId {
+        module_id,
+        member_id: function_id,
+    } = fun;
+    let ty_args = vec![];
+    let args = vec![];
+    let payload = TransactionPayload::EntryFunction(EntryFunction::new(
+        module_id,
+        function_id,
+        ty_args,
+        args,
+    ));
+    let transaction = TransactionBuilder::new(alice.clone())
+        .fee_payer(bob.clone())
+        .payload(payload)
+        .sequence_number(harness.sequence_number(alice.address()))
+        .max_gas_amount(1_000_000)
+        .gas_unit_price(1)
+        .sign_fee_payer();
+
+    let output = harness.run_raw(transaction);
+    assert_success!(*output.status());
+
+    let txn_ctx_store = harness
+        .read_resource::<crate::tests::transaction_context::TransactionContextStore>(
+            alice.address(),
+            parse_struct_tag("0x1::transaction_context_test::TransactionContextStore").unwrap(),
+        )
+        .unwrap();
+
+    let fee_payer_opt = txn_ctx_store.fee_payer_opt;
+    println!("fee_payer_opt: {:?}", fee_payer_opt);
```

Comment:
Are the print statements needed?

Reply:
Removed as they were for testing purpose.